src/ui/components/results/verticalresultscomponent.js

```diff
@@ -358,7 +363,8 @@ export default class VerticalResultsComponent extends Component {
       const resultsHeaderData = {
         resultsLength: this.results.length,
         resultsCount: this.resultsCount,
-        appliedFilterNodes: this.appliedFilterNodes,
+        removableFilterNodes: this.appliedFilterNodes,
+        irremovableFilterNodes: this.nlpFilterNodes,
```

Comment:
Minor nit, but I think we shouldn't leak the concept of removable or irremovable into VerticalResults. That's internal to the ResultsSectionHeader itself. As far as VerticalResults is concerned, it passes nlpFilters and appliedFilters into the ResultsHeader component. I think using hiddenFields in VerticalResults was a bit different because that would have caused VerticalResults to pass the ResultsSectionHeader unnecessary data.

Reply:
this is necessary because some FilterNodes passed to the ResultsHeader component (namely nlpFilters but could be anything) are not able to be removable.

Reply:
We could rename FilterMetadata.originComponent to just origin, change that to "nlpFilter" for nlpFilters, but I think it actually makes sense to leak this since the logic for removable/irremovable is inside of ResultsHeader (also it's just simpler in the current moment lol)

Reply:
ah okay, updated! that does feel kinda nice